code/go/0chain.net/smartcontract/dbs/event/users.go

```diff
@@ -0,0 +1,87 @@
+package event
+
+import (
+	"errors"
+	"fmt"
+
+	"0chain.net/smartcontract/dbs"
+	"gorm.io/gorm"
+)
+
+type User struct {
+	gorm.Model
+	TransactionID uint   `json:"transaction_id"`
+	Balance       int64  `json:"balance" msgpack:"b"`
+	UserID        string `json:"user_id" gorm:"uniqueIndex"`
+}
+
+func (edb *EventDb) GetUser(usrID string) (*User, error) {
+	var user User
+
+	result := edb.Store.Get().
+		Model(&User{}).
+		Where(&User{UserID: usrID}).
+		First(&user)
+	if result.Error != nil {
+		return nil, fmt.Errorf("error retrieving user %v, error %v",
+			usrID, result.Error)
+	}
+
+	return &user, nil
+}
+
+func (edb *EventDb) GetUsersForTxnID(txnID uint) (*[]User, error) {
+	var usrs []User
+	result := edb.Store.Get().
+		Model(&User{}).
+		Where(&User{TransactionID: txnID}).
+		Find(&usrs)
+	return &usrs, result.Error
+}
+
+func (edb *EventDb) overwriteUser(usr User) error {
+	result := edb.Store.Get().
+		Model(&User{}).
+		Where(&User{UserID: usr.UserID}).
+		Updates(&usr)
+	return result.Error
+}
+
+func (edb *EventDb) addOrOverwriteUser(usr User) error {
+	exists, err := usr.exists(edb)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return edb.overwriteUser(usr)
+	}
+
+	result := edb.Store.Get().Create(&usr)
+	return result.Error
+}
+
+func (edb *EventDb) UpdateUser(updates dbs.DbUpdates) error {
```

Comment:
is this being used?

Reply:
Yes, in the `emitEventDb` above (which will be changed to `emitEvent`